errai-data-binding/src/test/java/org/jboss/errai/databinding/client/TestModelWithIgnoreAnnotation.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jboss.errai.databinding.client;
+
+import java.util.Collection;
+import java.util.Map;
+
+import org.jboss.errai.databinding.client.api.Bindable;
+import org.jboss.errai.databinding.client.api.IgnoreBinding;
+
+/**
+ * Object besed on GXT ModelData intrface
+ * 
+ * @see https://docs.sencha.com/gxt/3.x/javadoc/com/sencha/gxt/legacy/client/data/ModelData.html
+ *
+ * @author Sašo Petrovič <saso.petrovic@gmail.com>
+ * 
+ */
+
+@Bindable
+public class TestModelWithIgnoreAnnotation {
+	
+	private Integer id;
+	private String name;
+	private String age;
+	
+	@IgnoreBinding
+	public <X> X get(String property) {
+		return null;
+	}
+
+	public Map<String, Object> getProperties() {
+		return null;
+	}
+
+	public <X> X remove(String property) {
+		return null;
+	}
+	
+	@IgnoreBinding
+	public void set(String property, Object value) {
+		return;
+	}
+	
+	@IgnoreBinding
+	public Integer getId() {

```

Comment:
Please see comment on `generateNonAccessorMethods`. We should write a test for this method that verifies that `getId` dispatches to the actual model and not the proxy. Since `setId` does not have the annotation, it will be properly proxied, so you could call `setId` with some value and verify that `getId` returns the same value.

Reply:
I've been trying to solve this issue but as far as I can tell this cannot be done without changing the bindable proxy generator. Because calling "ignored methods" on a proxy of a model does not reach the model, because the model is stored in "agent.target". 

So the generator should always create the methods, just not execute the "agent.updateWidgetsAndFireEvent".

But if the methods have to stay, I'm back at square one with my existing problem. Having two methods with the same signature (get("key"), set("key", "value")).

I could change the generators and leave the feature if you think it could be useful, but to fix my problem I would have to add a condition not to generate the BindableProxy specific methods.

Reply:
If you add `@IgnoreBinding` on a non-property method then I think it's fine for the data-binding generator to just ignore that method and generate nothing for it. That will fix your issue, since every method name that conflicts with the `BindableProxy` will not be a getter or setter.

But if you ignore an accessor, it should still be usable, just not as a property of the model (so it should still be proxied). If we could make that change, then I would be satisfied with this feature.

Another option, to enhance this, is we could add a property to the annotation to specify whether method should be ignored as an accessor, or fully ignored. I don't think this is necessary, but it's an option.

Reply:
I've assumed that method get("key") would be a accessor method, but I've tested the generator and it's not; is grabbed by non-property method generator.

But still, I don't see how to propagate the changes of a non-property method to the source. Here it the test I've been working on.  Because the method that is ignored is not proxied try's to access the property 'age' that is null, because the model value is in 'agetn.target'. 

![image](https://user-images.githubusercontent.com/26303578/30424963-df480374-9947-11e7-81c3-ed679ce2ad17.png)

![image](https://user-images.githubusercontent.com/26303578/30424978-e98c4890-9947-11e7-9d0a-b65bbc4472df.png)